### PR TITLE
Add debug print for intermediate values 

### PR DESCRIPTION
**Pull request by real-or-random**
solves https://github.com/sipa/bips/issues/188

Tell me if I overdid it. But I think it's still readable, in particular if you have syntax highlighting and the comments are displayed differently.

This also adjusts the variables to match the BIP text. 

I should have based this on top of #196, sorry. Let's first get #196 merged and I can rebase.

### CODE REVIEW

**Comment by real-or-random** [comment]
That sounds reasonable.

File: bip-0340/reference.py [hunk 1]
```diff
@@ -81,38 +90,43 @@ def is_square(x):
...
+        debug_print_vars()
         raise ValueError('The secret key must be an integer in the range 1..n-1.')
-    P = point_mul(G, x)
+    P = point_mul(G, d0)
     return bytes_from_point(P)
 
-def schnorr_sign(msg, seckey0, aux_rand):
+def schnorr_sign(msg, seckey, aux_rand):
     if len(msg) != 32:
+        debug_print_vars()
```
Comment (jonasnick) [inline bip-0340/reference.py]: I think this `debug_print_vars()` is not very helpful and can be removed to reduce lines of code. The main reason for having these is to allow comparing your implementation with the test vectors (presumably you only need one test vector). But none of the test vectors hit this case. If your implementation accepts `len(msg) != 32`, and you give the same inputs to the reference code, you will immediately see where the problem lies thanks to the exception alone. I think this holds for any `debug_print_vars` before raising exceptions.

File: bip-0340/reference.py [hunk 2]
```diff
@@ -62,7 +71,7 @@ def lift_x_square_y(b):
     y = pow(y_sq, (p + 1) // 4, p)
     if pow(y, 2, p) != y_sq:
         return None
-    return [x, y]
+    return (x, y)
```
Comment (jonasnick) [inline bip-0340/reference.py]: should be tuple in `lift_x_even_y` as well

File: bip-0340/reference.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def is_square(x):
     return pow(x, (p - 1) // 2, p) == 1
 
 def has_square_y(P):
-    return not is_infinity(P) and is_square(y(P))
+    return (not is_infinity(P)) and (is_square(y(P)))
```
Comment (jonasnick) [inline bip-0340/reference.py]: Putting parantheses`is_square(y(P))` in parantheses is unnecessary.

**Comment by real-or-random** [comment]
fixed, and added another commit that improves checking of the test vectors

**Comment by sipa** [comment]
ACK

**Pull request merged**